src/main/java/com/atlassian/db/replica/spi/ReplicaConsistency.java

```diff
@@ -12,12 +12,24 @@
 public interface ReplicaConsistency {
 
     /**
-     * Informs that {@code main} received an UPDATE, INSERT or DELETE.
+     * Informs that {@code main} received an UPDATE, INSERT or DELETE or transaction commit
+     * when in a transaction.
      *
      * @param main connects to the main database
      */
     void write(Connection main);
 
+    /**
+     * Invoked just before transaction commit.
+     *
+     * Notice: The method will not handle all writes. Writes done outside of a transaction
+     * needs to be handled in `ReplicaConnection#write`.
+     *
+     * @param main connects to the main database
+     */
+    default void preWrite(Connection main){
```

Comment:
Good that this is added in `SPI`, it would be risky in `API`

Reply:
Why?

Reply:
Adding methods to SPI is risky. Adding methods to API is not risky. 😉